packages/next/src/client/script.tsx

```diff
@@ -127,6 +145,12 @@ const loadScript = (props: ScriptProps): void => {
 
   el.setAttribute('data-nscript', strategy)
 
+  // Load styles associated with this script
+  // Useful when loading 3Ps with styles
+  if (stylesheets) {
+    insertStylesheets(stylesheets)
+  }
+
```

Comment:
Is there a reason to not load the stylesheet earlier than when the script loads? I would imagine any script related stylesheets would not be using global style rules that affect the document except for what the script itself injects/modifies. Seems like loading here is too late and may potentially race between script execution and style application. It'd probably be better to just render these stylesheets wtih the Script component using the precedence prop to opt into Float hoisting. The only problem with this strategy is that it would only work in App Router at the moment.